cli/azd/cmd/templates.go

```diff
@@ -390,13 +407,14 @@ func (a *templateSourceAddAction) Run(ctx context.Context) (*actions.ActionResul
 		}
 
 		// Validate the custom source config
-		_, err := a.sourceManager.CreateSource(ctx, sourceConfig)
+		_, err := a.sourceManager.CreateSource(ctx, sourceConfig, a.console)
```

Comment:
IMHO - A template source should not need a reference to our console... If the template source needs to write a message it should throw an error and have the consumer (this cmd) handle the console writing.

Reply:
I'm not sure whether we should start an interactive session to login to the github CLI here.  I'd prefer we fail with an error message prompting them to login to the gh CLI and then run the command again.  It isn't like we are in the middle of a long running process.

Reply:
azd handles gh-cli automatically. We never ask folks to manually log in to gh-cli first.
The gh-cli is automatically installed in ~/.azd/bin and we start the login flow for it during azd pipeline config (if not already logged in)